app/src/androidTest/java/org/simple/clinic/overdue/AppointmentRepositoryAndroidTest.kt

```diff
@@ -875,4 +877,77 @@ class AppointmentRepositoryAndroidTest {
       assertThat(it.appointmentType).isEqualTo(Automatic)
     }
   }
+
+  @Test
+  fun when_picking_overdue_appointment_then_the_latest_recorded_bp_should_be_considered() {
+    val facility = testData.qaFacility()
+    val now = Instant.now(clock)
+
+    fun createBloodPressure(patientProfile: PatientProfile, recordedAt: Instant, updatedAt: Instant): BloodPressureMeasurement {
+      return testData.bloodPressureMeasurement(
+          patientUuid = patientProfile.patient.uuid,
+          recordedAt = recordedAt,
+          updatedAt = updatedAt
+      )
+    }
+
+    fun createPatient(fullName: String): PatientProfile {
+      return testData.patientProfile(generatePhoneNumber = true)
+          .let { patientProfile ->
+            patientProfile.copy(patient = patientProfile.patient.copy(fullName = fullName))
+          }
+    }
+
+    fun scheduleAppointment(patientProfile: PatientProfile): Single<Appointment> {
+      return appointmentRepository.schedule(
+          patientUuid = patientProfile.patient.uuid,
+          appointmentDate = LocalDate.now(clock).minus(2, ChronoUnit.DAYS),
+          appointmentType = Manual,
+          currentFacility = facility)
+    }
+
+    val patient1 = createPatient("Patient with one latest updated retro-active BP")
+    val patient2 = createPatient("Patient with no retro-active BP")
+
+    patientRepository.save(listOf(patient1, patient2)).blockingAwait()
+
+    val bpOneForPatient1 = createBloodPressure(
+        patientProfile = patient1,
+        recordedAt = now,
+        updatedAt = now
+    )
+    val bpTwoForPatient1 = createBloodPressure(
+        patientProfile = patient1,
+        recordedAt = now.minus(10, ChronoUnit.DAYS),
+        updatedAt = now.plusSeconds(10)
+    )
+
+    val bpOneForPatient2 = createBloodPressure(
+        patientProfile = patient2,
+        recordedAt = now,
+        updatedAt = now)
+
+    val bpTwoForPatient2 = createBloodPressure(
+        patientProfile = patient2,
+        recordedAt = now.plus(10, ChronoUnit.MINUTES),
+        updatedAt = now
+    )
+
+    bpRepository.save(listOf(bpOneForPatient1, bpTwoForPatient1, bpOneForPatient2, bpTwoForPatient2)).blockingAwait()
+
+    scheduleAppointment(patient1).blockingGet()
+    scheduleAppointment(patient2).blockingGet()
+
+    val appointments = appointmentRepository.overdueAppointments(facility).blockingFirst().associateBy { it.fullName }
```

Comment:
We can simplify this test without the `Map` like this?
```kotlin
    val appointments = appointmentRepository.overdueAppointments(facility).blockingFirst()

    assertThat(appointments.size).isEqualTo(2)
    assertThat(appointments[0].bloodPressure).isEqualTo(bpTwoForPatient2)
    assertThat(appointments[1].bloodPressure).isEqualTo(bpOneForPatient1)
```

Reply:
We cannot determine the order of appointments received in `appointmentRepository.overdueAppointments(facility)` since there are multiple factors that affect this order.
Also, the reason for using `fullName` is so that the test case is more readable.